app/services/PreferenceService.ts

```diff
@@ -0,0 +1,58 @@
+import { PrismaClient } from "@prisma/client";
+import UnprocessableEntityException from "../exceptions/UnprocessableEntityException";
+
+const prisma = new PrismaClient();
+
+export const PreferenceService = {
+  /**
+   * Updates user preference database
+   *
+   * @param uid user id
+   * @param data data to update the information
+   * @return {Promise<JSON>} same data that is inputed
+   * @throws {UnprocessableEntityException} Data is unprocessable
+   */
+  async update(uid: string, data: JSON): Promise<JSON> {
+    try {
+      //@ts-ignore
+      const pref = data.preference;
```

Comment:
Same comment as I did on the AuthService, define a custom data type to match preferences and use that for the return data type, and avoid using @ts-ignore.